include/alpaka/atomic/Op.hpp

```diff
@@ -38,8 +38,15 @@ namespace alpaka
                 {
                     auto const old(*addr);
                     auto & ref(*addr);
+#if BOOST_COMP_GNUC 
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wconversion"
```

Comment:
alpaka atomic operations support types smaller than the size of int. In that case the compiler is always warning about type conversions.
This PR is ignoring the error at this place but in general we need to take care which types can be used for atomic operations. 
This depends on the alpaka backend and should be handled in another PR.